jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -39,47 +36,33 @@ pointer_typet select_pointer_typet::convert_pointer_type(
 pointer_typet select_pointer_typet::specialize_generics(
   const pointer_typet &pointer_type,
   const generic_parameter_specialization_mapt
-    &generic_parameter_specialization_map,
-  generic_parameter_recursion_trackingt &visited_nodes) const
+    &generic_parameter_specialization_map) const
 {
   auto parameter = type_try_dynamic_cast<java_generic_parametert>(pointer_type);
   if(parameter != nullptr)
   {
-    const irep_idt &parameter_name = parameter->get_name();
+    irep_idt parameter_name = parameter->get_name();
```

Comment:
No need for this to change? `const irep_idt &` is pretty widely used for a const `irept`'s data, this probably isn't the place to fight that battle

Reply:
I assign to it at the bottom of the loop.